test-data/unit/check-isinstance.test

```diff
@@ -391,7 +391,7 @@ def f(x: Union[List[int], List[str], int]) -> None:
     else:
         x[0] # E: Value of type "int" is not indexable
         x + 1
-    x[0] # E: Value of type "int" is not indexable
+    x[0] # E: Value of type "Union[List[int], List[str], int]" is not indexable
```

Comment:
I'm guessing this is because of the better determination of `original_type` right? I approve of the new error message.